map_model/src/pathfind/v2.rs

```diff
@@ -3,82 +3,181 @@
 //! things here will probably move into pathfind/mod.rs.
 
 use anyhow::Result;
+use serde::{Deserialize, Serialize};
+
+use geom::Duration;
 
 use crate::pathfind::uber_turns::UberTurnV2;
-use crate::{DirectedRoadID, Map, Path, PathConstraints, PathRequest, PathStep, TurnID, UberTurn};
+use crate::{
+    DirectedRoadID, Map, MovementID, Path, PathConstraints, PathRequest, PathStep, TurnID, UberTurn,
+};
+
+/// One step along a path.
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub enum PathStepV2 {
+    /// Original direction
+    Along(DirectedRoadID),
+    /// Opposite direction, sidewalks only
+    Contraflow(DirectedRoadID),
+    Turn(MovementID),
+}
 
-/// Transform a sequence of roads representing a path into the current lane-based path, by picking
-/// particular lanes and turns to use.
-pub fn path_v2_to_v1(
+/// A path between two endpoints for a particular mode. This representation is immutable and doesn't
+/// prescribe specific lanes and turns to follow.
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub struct PathV2 {
+    steps: Vec<PathStepV2>,
+    // TODO There will be a PathRequestV2, but I'm not sure how it'll change yet.
     req: PathRequest,
-    road_steps: Vec<DirectedRoadID>,
```

Comment:
I moved this code down to `to_v1` below and made it operate on `PathStepV2` instead of pairs of directed roads